galley/pkg/server/args.go

```diff
@@ -83,6 +83,9 @@ type Args struct {
 	// allows Galley to start when not all supported CRD are
 	// registered with the kube-apiserver.
 	DisableResourceReadyCheck bool
+
+	// ForceStop is set as true when used for testing to make the server stop quickly
+	ForceStop bool
```

Comment:
I am a bit worried about this. I don't understand why we need it. I'd have expected the close/reset of downstream components such as Mixer and Pilot should allow graceful stop to close quickly. Time-wise, what is the difference you're observing with ForceStop?

Reply:
Yep, Oz, this is my expectation as well. However there are two issues currently which blocks galley from stopping:
1.  mixer has not stopped galley client in his code, hope this PR : https://github.com/istio/istio/pull/10772 can fix this. -- waiting review comments now.

2. Since galley, mixer and pilot are all system scope components in our test framework by default, so it will be closed when one test finished running.  This means the mixer's close will not be called and it  causes the galley client in mixer has not been closed. -- THis brings galley fails to stop.

Reply:
I see. Ideally, we should restart Mixer & Pilot in every test run (for the native case), unless there is a compelling performance need. That will be a much simpler way to avoid cross-test interference.

For the short term, can you implement this as a separate method instead of an argument? And please also make sure to add enough comments on it to indicate this is for testing only, and should not be depended on, similar to the comment you have in the arguments.

Having it as a method will make it very clear in code where we need it. (i.e. something like "ForceStop()" method on the server object).

Thanks!

Reply:
OK, to align with Close() method, I have used ForceClose() as method name